plugins/scss/index.js

```diff
@@ -154,14 +159,54 @@ const writer = function(thisPackage, outputDir) {
       file = path.resolve(outputDir, 'combined.scss');
       args.push(file);
       console.log('Writing ' + file);
-      promises.push(concat(scssEntries, file));
+      promises.push(concat.concatAsync(scssEntries, file));
     }
 
     file = path.resolve(outputDir, 'node-sass-args');
     console.log('Writing ' + file);
     promises.push(fs.writeFileAsync(file, args.join(' ')));
 
-    return Promise.all(promises);
+    return new Promise(function(resolve, reject) {
+      resolve(Promise.all(promises).then(function() {
+        var promises = [];
+        // Write a file for each theme also
+        if (utils.isAppPackage(thisPackage) &&
+            thisPackage.build.scss &&
+            thisPackage.build.themes &&
+            thisPackage.build.themes.length) {
+          // Take off the combined.scss generic file
+          args.pop();
+
+          var themeDir = outputDir + '/themes';
+          mkdirp.sync(themeDir);
```

Comment:
It is considered poor practice in Node to use synchronous calls. Please consider using the async call instead. You may want to promisify it similarly to how `concat` was done above.

Reply:
I'm not sure if theres an async method? https://github.com/substack/node-mkdirp/blob/master/index.js